types/mui-datatables/index.d.ts

```diff
@@ -167,7 +174,7 @@ export interface MUIDataTableColumnState extends MUIDataTableColumnOptions {
 
 export interface MUIDataTableColumnOptions {
     customBodyRender?: (
-        value: any,
+        value: string,
```

Comment:
@Patrick-Erichsen Could you please explain this change? My team has been using the `customBodyRender` function successfully thus far with `number` and `boolean` values. 

See this example from the mui-datatables repo, where between Lines 60-98, the column options are found for the **Salary** and **Active** columns. The **Salary** column contains number values, and the **Active** column contains boolean values.
Also note on Line 91 that the `updateValue()` function can receive a non-string type value (in this case, a boolean).
https://github.com/gregnb/mui-datatables/blob/master/examples/component/index.js#L60-L98

Reply:
Thanks for taking a closer look at these updates and linking those examples! I'm AFK this weekend but will give a more thorough response Monday.

When making these type changes, I was attempting to pass an object to `customBodyRender` and getting a `.toString()` version of my object passed as a param. I ended up stringifying/parsing my object to pass it around. 

I didn't see those examples previously since I hadn't tried passing anything except objects/strings, so I incorrectly assumed that strings were the only valid param.

My guess is that only primitives are allowed? Will dig into it more Monday though.

Reply:
Hey Daniel, sorry for the delay in getting back to you on this. After some digging, this seems like purely a user error on my part. Really not sure where I got this mixed up, so thanks again for pointing it out.

I believe the correct type for that param on `customBodyRender` should be [`MUIDataTableData['data'][number]`](https://github.com/Patrick-Erichsen/DefinitelyTyped/blob/dfe100fab161218fdff903e270984dd46d6537a4/types/mui-datatables/index.d.ts#L28)

```ts
export interface MUIDataTableData {
    data: Array<object | number[] | string[]>;
    index: number;
}

...

export interface MUIDataTableColumnOptions {
    customBodyRender?: (
        value: MUIDataTableData['data'][number],
        tableMeta: MUIDataTableMeta,
        updateValue: (value: string) => void,
    ) => string | React.ReactNode;
    ....
}
```

If this looks good to you, let me know and I can open up [this pull request I have ready](https://github.com/DefinitelyTyped/DefinitelyTyped/compare/master...Patrick-Erichsen:pe-mui-datatables-fix?expand=1#diff-13e6d38691187d88542504b110fd099bR177).

Reply:
@russelldc

Reply:
Hey @Patrick-Erichsen , sorry it took so long to get back to you on this -- your response had coincided with my vacation!

Yeah, I think it would be best to simply revert it back to `any`.
I came to the same conclusion after digging through the source, examples, docs and GH issues.


I also found this in the docs just now, for what it's worth:
https://github.com/gregnb/mui-datatables/blame/master/README.md#L315
```js
// `customBodyRender` is called with these arguments:

function(value: any, tableMeta: {
  rowIndex: number,
  ...
```

Reply:
Hey @russselldc, not a problem. I'm okay with reverting the prop back to `any`, and I've opened a PR to do so here -> https://github.com/DefinitelyTyped/DefinitelyTyped/pull/49048.